extension-style/src/main/java/com/mapbox/maps/extension/style/layers/LayerExt.kt

```diff
@@ -0,0 +1,63 @@
+package com.mapbox.maps.extension.style.layers
+
+import com.mapbox.maps.LayerPosition
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.extension.style.StyleInterface
+
+/**
+ * Bind the layer to the map controller persistently.
+ *
+ * Note: This is an experimental API and is a subject to change.
+ *
+ * Whenever a new style is being parsed and currently used style has persistent layers,
+ * an engine will try to do following:
+ *   - keep the persistent layer at its relative position
+ *   - keep the source used by a persistent layer
+ *   - keep images added through `addStyleImage` method
+ *
+ * In cases when a new style has the same layer, source or image resource, style's resources would be
+ * used instead and `MapLoadingError` event will be emitted.
+ *
+ * @param style The style
+ * @param position the position that the current layer is added to
+ */
+@MapboxExperimental
+internal fun Layer.bindPersistentlyTo(style: StyleInterface, position: LayerPosition? = null) {
```

Comment:
@tobrun it seems we can make this function as internal and only expose the `addPersistentLayer` API for the `StyleInterface`, this should align better with the gl-native design choice.